render.c

```diff
@@ -19,6 +22,45 @@ static void draw_rect(cairo_t *cairo, struct slurp_box *box, uint32_t color) {
 			box->width, box->height);
 }
 
+static void draw_rect_label(cairo_t *cairo, struct slurp_box *box, enum slurp_label_anchor anchor,
+		uint32_t text_color, uint32_t background_color, const char *font_family, const char *text) {
+	if (anchor == ANCHOR_NONE)
+		return;
+	cairo_select_font_face(cairo, font_family,
+						 CAIRO_FONT_SLANT_NORMAL,
+						 CAIRO_FONT_WEIGHT_NORMAL);
+	cairo_set_font_size(cairo, 20);
+	cairo_text_extents_t extents;
+	cairo_text_extents(cairo, text, &extents);
+	struct slurp_box labelbox = { 0 };
+	labelbox.width = ceil(extents.width) + 2*LABEL_BOX_PADDING;
+	labelbox.height = ceil(extents.height) + 2*LABEL_BOX_PADDING;
+	labelbox.x = box->x + (box->width - labelbox.width) / 2;
+	labelbox.y = box->y + (box->height - labelbox.height) / 2;
+
+	if (anchor & ANCHOR_TOP && anchor & ANCHOR_BOTTOM) {
+		// Nothing to do
+	} else if (anchor & ANCHOR_TOP) {
+		labelbox.y = box->y + 20;
+	} else if (anchor & ANCHOR_BOTTOM) {
+		labelbox.y = box->y + box->height - 20 - labelbox.height;
+	}
+	if (anchor & ANCHOR_LEFT && anchor & ANCHOR_RIGHT) {
+		// Nothing to do
+	} else if (anchor & ANCHOR_LEFT) {
+		labelbox.x = box->x + 10;
```

Comment:
why is the horizonal margin different than the vertical margin?

Reply:
I've used the same offset as used for the dimensions.